JedisManager.cfc

```diff
@@ -70,11 +78,19 @@ component {
             // Use the Jedis resource
             jedis.setex(arguments.cacheKey, arguments.cacheDurationInSeconds, arguments.dataToCache);
         
+        } catch ( "JedisManager.retriveJedis.resource.error" e ) {
+            rethrow;
         } catch (Exception e) {
-            throw(message="JedisManager insert cache error: "&e.message, detail=e.detail);
+            throw(
+                type   = "JedisManager.cacheInsert.error",
+                message= "JedisManager insert cache error: "&e.message,
+                detail = e.detail
+            );
         } finally {
             // Return the Jedis resource to the pool
-            returnJedisResource(jedis);
+            if(isDefined("jedis")){
```

Comment:
Is `isDefined` the best option here?

I don't have a better suggestion right away, but maybe `!isNull()` or similar would be better? Not sure.

Reply:
(same for the other `isDefined` instances)